### PR TITLE
[2.8][Translation] added message cache.

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | #13676 |
| Tests pass? | yes |
| License | MIT |

``` yml
# app/config/config.yml
framework:
    translator:
        cache: translation.cache.default
```

The message cache Interface contains now only one method:

``` php
interface MessageCacheInterface
{
    public function cache($locale, array $options = array());
}
```

for more details see #13986.

### CODE REVIEW

File: src/Symfony/Component/Translation/MessageCache.php [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ */
+
+namespace Symfony\Component\Translation;
+
+use Symfony\Component\Config\ConfigCacheInterface;
+use Symfony\Component\Config\ConfigCacheFactoryInterface;
+use Symfony\Component\Config\ConfigCacheFactory;
+
+/**
+ * @author Abdellatif Ait Boudad <a.aitboudad@gmail.com>
```
Comment (pborreli) [inline src/Symfony/Component/Translation/MessageCache.php]: boudad is lowercase :laughing:

Comment (aitboudad) [inline src/Symfony/Component/Translation/MessageCache.php]: :sleeping:

Comment (aitboudad) [inline src/Symfony/Component/Translation/MessageCache.php]: fixed thanks

Comment (cordoval) [inline src/Symfony/Component/Translation/MessageCache.php]: on his own last name nice @pborreli

**Comment by mpdude** [comment]
Why do you think do we need another interface for this and make the `Translator` aware of it?

I mean, why can't we just decorate the `Translator` with a caching layer that keeps the results of `trans()` and `transChoice()` in (shared) memory?

**Comment by aitboudad** [comment]
@mpdude The current implementation try to simplify the `Translator` itself and to not be aware of how to caching and also it provide the consistant way to override `MessageCatalogue`.
Decorating solve the issue but it isn't ideal as it's attempt to not call `trans()` and `transChoice()` of the `Translator` which can make hard to handle other decorated class like `LoggingTranslator` or `DataCollectorTranslator` for example.

**Comment by aitboudad** [comment]
closed in favor of #14671.

**Pull request closed without merging**